frontend/src/pages/modelCatalog/screens/RegisterCatalogModel.tsx

```diff
@@ -0,0 +1,216 @@
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  FormGroup,
+  PageSection,
+  Form,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import React from 'react';
+import { Link, useNavigate, useParams } from 'react-router';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import { ModelRegistrySelectorContext } from '~/concepts/modelRegistry/context/ModelRegistrySelectorContext';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import {
+  ModelLocationType,
+  useRegisterCatalogModelData,
+} from '~/pages/modelRegistry/screens/RegisterModel/useRegisterModelData';
+import RegistrationCommonFormSections from '~/pages/modelRegistry/screens/RegisterModel/RegistrationCommonFormSections';
+import {
+  isNameValid,
+  isRegisterCatalogModelSubmitDisabled,
+  registerModel,
+} from '~/pages/modelRegistry/screens/RegisterModel/utils';
+import { SubmitLabel } from '~/pages/modelRegistry/screens/RegisterModel/const';
+import { ModelDetailsRouteParams } from '~/pages/modelCatalog/const';
+import { modelDetailsUrl } from '~/pages/modelCatalog/routeUtils';
+import RegisterModelDetailsFormSection from '~/pages/modelRegistry/screens/RegisterModel/RegisterModelDetailsFormSection';
+import RegistrationFormFooter from '~/pages/modelRegistry/screens/RegisterModel/RegistrationFormFooter';
+import { registeredModelUrl } from '~/pages/modelRegistry/screens/routeUtils';
+import { useAppSelector } from '~/redux/hooks';
+import { ModelCatalogContext } from '~/concepts/modelCatalog/context/ModelCatalogContext';
+import { CatalogModel } from '~/concepts/modelCatalog/types';
+import ModelRegistrySelector from '~/pages/modelRegistry/screens/ModelRegistrySelector';
+import useModelRegistryAPIState from '~/concepts/modelRegistry/context/useModelRegistryAPIState';
+import { decodeParams, findModelFromModelCatalogSources } from '~/pages/modelCatalog/utils';
+import {
+  ModelRegistryCustomProperties,
+  ModelRegistryMetadataType,
+} from '~/concepts/modelRegistry/types';
+
+const RegisterCatalogModel: React.FC = () => {
+  const navigate = useNavigate();
+  const params = useParams<ModelDetailsRouteParams>();
+  const decodedParams = decodeParams(params);
+
+  const { preferredModelRegistry } = React.useContext(ModelRegistrySelectorContext);
+  const { modelCatalogSources } = React.useContext(ModelCatalogContext);
+
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [submitError, setSubmitError] = React.useState<Error | undefined>(undefined);
+  const [formData, setData] = useRegisterCatalogModelData();
+  const [registeredModelName, setRegisteredModelName] = React.useState<string>('');
+  const [versionName, setVersionName] = React.useState<string>('');
+  const [errorName, setErrorName] = React.useState<string | undefined>(undefined);
```

Comment:
Hey - it's ok not to block this PR on this since I know this is duplicated from the RegisterModel and RegisterVersion pages, but I think this state is a little confusing. I know it exists because when there's an error submitting we need to show the names of the resources that were attempted to be created even if the form has since changed, but with these names it looks like duplicated state that we might try to use expecting it to be the form values. I would love to rename these to something like `submittedRegisteredModelName` and `submittedVersionName`, and maybe `errorName` (and the `ErrorName` type) should be `registrationErrorType` or something.

We probably also want to see what we can deduplicate between these 3 pages... that definitely belongs in a followup tech debt issue though.